Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift

```diff
@@ -62,14 +62,26 @@ final class BuildPhaseGenerator: BuildPhaseGenerating {
                              pbxTarget: PBXTarget,
                              fileElements: ProjectFileElements,
                              pbxproj: PBXProj,
-                             sourceRootPath _: AbsolutePath) throws {
+                             sourceRootPath: AbsolutePath) throws {
         if let headers = target.headers {
             try generateHeadersBuildPhase(headers: headers,
+                                          includePublicFiles: target.modulemap == nil,
                                           pbxTarget: pbxTarget,
                                           fileElements: fileElements,
                                           pbxproj: pbxproj)
         }
 
+        if let modulemap = target.modulemap {
```

Comment:
I believe the criteria for including the `generateCopyHeadersBuildPhase` is when `!target.supportsHeadersPhase`.

e.g. A normal `.framework` allows specifying a `modulemap` and does not require any special copy, Xcode automatically deals with that.